### PR TITLE
SCMOD-11932: Added package locks

### LINKED ISSUE #28: Images consuming this with a zypper update removes the java.security modifications
If another image consumes this e.g. [`tomcat-image`](https://github.com/CAFapi/opensuse-tomcat-image/blob/develop/opensuse-tomcat-image/src/main/docker/Dockerfile) is built with a `zypper update` and there has been a java release since, then the `zypper update` will update the java version. The problem is that the `java.security` file in this image is patched and if a consuming component updates java via `zypper update` then the security file updates are wiped out.

### PR DESCRIPTION
**Pull request by michael-bryson**
JIRA: https://portal.digitalsafe.net/browse/SCMOD-11932

### CODE REVIEW

**Comment by michael-bryson** [comment]
@dermot-hardy Tested in the tomcat image build and I can see the following in the log:
```
DOCKER> Step 22/36 : FROM docker.io/cafinternal/prereleases:opensuse-jre8-3.2.0-SCMOD-11932-SNAPSHOT
DOCKER> 
DOCKER> ---> 9122ab9d7801
DOCKER> Step 23/36 : RUN zypper -n refresh &&     zypper -n update
DOCKER> 
DOCKER> ---> Running in 87bcc408ef94
DOCKER> Repository 'Non-OSS Repository' is up to date.
DOCKER> Repository 'Main Repository' is up to date.
DOCKER> Repository 'Main Update Repository' is up to date.
DOCKER> Repository 'Update Repository (Non-Oss)' is up to date.
DOCKER> All repositories have been refreshed.
DOCKER> Loading repository data...
DOCKER> Reading installed packages...
DOCKER> The following item is locked and will not be changed by any action:
 Installed:
  java-1_8_0-openjdk

Nothing to do.
```

**Comment by hoshposh** [comment]
@michael-bryson What does this do? I do not have access to the JIRA system so cannot read the issue.

**Comment by michael-bryson** [comment]
@hoshposh We discovered an issue where if another image built on top of this e.g. tomcat-image is built and there has been a java release since then the zypper update will update the java version. The problem we have is that we patch the java.security file in this image and if a consuming component updates the java via zypper then the security file updates are wiped out.

**Comment by hoshposh** [comment]
@michael-bryson Thanks for the info, and that is a nice solution. Cheers! 👍

File: src/main/docker/Dockerfile.jdk [hunk 1]
```diff
@@ -26,7 +26,8 @@ FROM ${DOCKER_HUB_PRIVATE}/cafinternal/prereleases:opensuse-base-2.2.0-SNAPSHOT
 # Refresh the OS repositories and install OpenJDK 8 Development Kit
 RUN zypper -n refresh && \
     zypper -n update && \
-    zypper -n install java-1_8_0-openjdk-devel
+    zypper -n install java-1_8_0-openjdk-devel && \
+    zypper al java-1_8_0-openjdk
```
Comment (dermot-hardy) [inline src/main/docker/Dockerfile.jdk:30]: Should this be `java-1_8_0-openjdk-devel`?

Comment (michael-bryson) [reply src/main/docker/Dockerfile.jdk:30]: `java-1_8_0-openjdk` covers the devel one too as well as other that start with `java-1_8_0-openjdk`, for example:
```
550fa87369cf:/ # zypper update
Loading repository data...
Reading installed packages...

The following item is locked and will not be changed by any action:
 Installed:
  java-1_8_0-openjdk

The following 3 package updates will NOT be installed:
  java-1_8_0-openjdk java-1_8_0-openjdk-devel java-1_8_0-openjdk-headless

The following NEW package is going to be installed:
  libyaml-cpp0_6

The following 54 packages are going to be upgraded:
  aaa_base ca-certificates-mozilla curl diffutils file file-magic findutils glibc krb5 libasound2 libblkid1 libcom_err2 libcurl4 libfdisk1 libfreebl3 libfreetype6 libgcc_s1 libgnutls30 libjpeg8 libldap-2_4-2 libldap-data libmagic1
  libmount1 libopenssl1_1 libpq5 libprocps7 libproxy1 libsmartcols1 libsoftokn3 libsolv-tools libstdc++6 libsystemd0 libudev1 libuuid1 libX11-6 libX11-data libxcb1 libxml2-2 libz1 libzypp mozilla-nspr mozilla-nss mozilla-nss-certs
  openssl-1_1 openSUSE-release openSUSE-release-appliance-docker pam permissions postgresql postgresql10 procps sysuser-shadow util-linux zypper

The following product is going to be reinstalled:
  "openSUSE Leap 15.1"

The following 2 packages require a system reboot:
  glibc libopenssl1_1

54 packages to upgrade, 1 new.
Overall download size: 22.1 MiB. Already cached: 0 B. After the operation, additional 201.3 KiB will be used.

    Note: System reboot required.
```

Comment (michael-bryson) [reply src/main/docker/Dockerfile.jdk:30]: Having said that I just tested `java-1_8_0-openjdk-devel` and it works the same way so either can be used.

**Review by dermot-hardy — APPROVED** [review]

**Pull request merged**